internal/keepers/factory.go

```diff
@@ -12,32 +12,47 @@ import (
 
 const maxObservationLength = 1_000
 
+var _ types.ReportingPluginFactory = (*keepersReportingFactory)(nil)
```

Comment:
curious, what does this statement do? The blank variable can't be used anywhere?

Reply:
This is the interface compatibility check

Reply:
It ensures the keepersReportingFactory implements the `types.ReportingPluginFactory`. If not, you can be assured the type system will catch error.